carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -41,8 +41,8 @@ namespace cpp_message
         mobility_operation_message_sub_=nh_->subscribe<>("outgoing_plain_mobility_operation_message",5, &Message::outbound_mobility_operation_message_callback,this);
         mobility_response_message_pub_=nh_->advertise<cav_msgs::ByteArray>("incoming_mobility_response_message_decoded",5);
         mobility_response_message_sub_=nh_->subscribe<>("outgoing_plain_mobility_operation_message",5, &Message::outbound_mobility_response_message_callback,this);
-        mobility_path_message_pub_=nh_->advertise<cav_msgs::ByteArray>("inbound_mobility_path_message",5);
-        mobility_path_message_sub_=nh_->subscribe<>("outbound_plain_mobility_path_message",5, &Message::outbound_mobility_path_message_callback,this);
+        mobility_path_message_pub_=nh_->advertise<cav_msgs::MobilityPath>("inbound_mobility_path_message",5);
```

Comment:
If you change your publisher type you almost certainly need to also change code where the publication occurred. I'm surprised it isn't throwing a build error.

Reply:
I did check that, I'm publishing the decoded output, which is already a MobilityPath msg. It wouldn't have worked with ByteArray.

Reply:
Ok good